src/pages/Auth/LoginPage/index.tsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import sha256 from 'utils/ts/SHA-256';
+import styles from './LoginPage.module.scss';
+import useLogin from '../../../utils/hooks/useLogin';
+
+interface IClassUser {
+  userId: HTMLInputElement | null
+  password: HTMLInputElement | null
+}
+
+const emailLocalPartRegex = /^[a-z_0-9]{1,12}$/;
+
+function LoginPage() {
+  const loginRef = React.useRef<IClassUser>({
+    userId: null,
+    password: null,
+  });
+  const [isAutoLoginFlag, setIsAutoLoginFlag] = React.useState(false);
+  const postLogin = useLogin({ isAutoLoginFlag });
+  const onToggleAutoLoginFlag = React.useCallback(() => {
+    setIsAutoLoginFlag(!isAutoLoginFlag);
+  }, [isAutoLoginFlag]);
+  const onSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const { userId, password } = loginRef.current;
+    if (userId === null) {
+      return;
+    }
+    if (password === null) {
+      return;
+    }
+    emailLocalPartRegex.test(userId.value);
+    const hashedPassword = await sha256(password?.value);
+
+    postLogin.mutate({
+      portal_account: userId.value,
+      password: hashedPassword,
+    });
+  };
+
+  return (
+    <div className={styles.template}>
+      <form className={styles.loginform} onSubmit={onSubmit}>
+        <input
+          ref={(inputRef) => { loginRef.current.userId = inputRef; }}
+          className={styles.styledinput}
+          autoComplete="username"
+          name="userId"
+          placeholder="아이디를 입력하세요"
+        />
+        <input
+          ref={(inputRef) => { loginRef.current.password = inputRef; }}
+          className={styles.styledinput}
+          type="password"
+          autoComplete="current-password"
+          name="password"
+          placeholder="비밀번호를 입력하세요"
+        />
+        <button type="submit" className={styles.loginform__button}>
+          로그인
+        </button>
+      </form>
+      <div className={styles.autologin}>
+        <label className={styles.autologin__label} htmlFor="autoLoginCheckBox">
+          <input className={styles.autologin__label__checkbox} checked={isAutoLoginFlag} onChange={onToggleAutoLoginFlag} type="checkbox" id="autoLoginCheckBox" />
+          <span className={styles.autologin__label__oncheck} />
```

Comment:
```suggestion
          <span role="checkbox" aria-checked={isAutoLoginFlag} className={styles.autologin__label__oncheck} />
```
접근성 향상을 위해 실제 checkbox 역할을 하는 span 태그에 속성을 추가해주어야 할 것 같네요.

Reply:
해당 부분에서는 [해당 lint 오류](jsx-a11y/control-has-associated-label)가 발생합니다. 이 부분은 제가 정확히 어떤 것이 옳은지 잘 모르겠습니다. 접근성향상이라고 한다는 것이 저 코드 추가로 인해 정확히 어떤 부분이 향상되는 건가요 기능상에서 지금 현재 더 개선되는 점이 있는지 궁금합니다

Reply:
일단 className 스타일의 경우에는 수정되었습니다. 한번 확인 부탁드립니다.
[a3fbc74](https://github.com/BCSDLab/KOIN_WEB_RECODE/pull/7/commits/a3fbc74d64a36e5617dbeb78eacaf73bc58f3ee4)
[1725017](https://github.com/BCSDLab/KOIN_WEB_RECODE/pull/7/commits/17250177443da5f7120a05a19b9f37ddbca01d18)
[8abd4a1](https://github.com/BCSDLab/KOIN_WEB_RECODE/pull/7/commits/8abd4a16436763e49e3e04c9aa2ccf51b10cd8cd)

Reply:
[해당 input checkbox에서 내장된 의미를 담당하고 있으므로 따로 aria-checked를 주지 않아도 됩니다.](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-checked) 그래서 쓸모없는 span태그를 aria-hidden으로 인식하지 않는 방향으로 제안하였습니다.
해당 제안은 필요없는 제안으로 보입니다.
해당 태그는 input태그를 없애고 따로 span태그만 활용할 경우 필요한 태그입니다.

Reply:
checkbox ui인 span과 input이 공존해서 복잡해져서 헷갈렸네요.
input의 display를 유지하며 이를 개선한 PR(#8) 올렸으니 확인해주세요~

Reply:
nice assist